refactor: Added name for background task started on enterBackground

antohisorin opened the pull request:
<!---
Thanks for contributing to the Amplitude iOS/tvOS/macOS SDK! 🎉

Please fill out the following sections to help us quickly review your pull request.
--->

### Summary

Added name for background task started on enterBackground to improve debugging.

### Checklist

* [X] Does your PR title have the correct [title format](https://github.com/amplitude/Amplitude-iOS/blob/main/CONTRIBUTING.md#pr-commit-title-conventions)?
* [X] Does your PR have a breaking change?:  NO

antohisorin commented:
@Mercy811 @qingzhuozhen is there a way we can land this MR? We need to track background tasks and what happens to them in our app. We don't want to fork the amplitude SDK only for this small change.

izaaz commented on Sources/Amplitude/Amplitude.m:
```diff
@@ -1250,7 +1250,7 @@ - (void)enterBackground {
     // Stop uploading
     [self endBackgroundTaskIfNeeded];
-    _uploadTaskID = [app beginBackgroundTaskWithExpirationHandler:^{
+    _uploadTaskID = [app _uploadTaskID = [app beginBackgroundTaskWithName:@"Amplitude: enterBackground" expirationHandler::^{
```
@antohisorin Looks like this does not compile? Shoudl this be

```
    _uploadTaskID = [app beginBackgroundTaskWithName:@"Amplitude: enterBackground" expirationHandler:^{
```

antohisorin replied:
Updated.

izaaz commented:
> @Mercy811 @qingzhuozhen @justin-fiedler is there a way we can land this MR? We need to track background tasks and what happens to them in our app. We don't want to fork the amplitude SDK only for this small change.

Sure @antohisorin. I can help get this merged and release a new version. Dropped a comment

antohisorin commented:
> > @Mercy811 @qingzhuozhen @justin-fiedler is there a way we can land this MR? We need to track background tasks and what happens to them in our app. We don't want to fork the amplitude SDK only for this small change.
> 
> Sure @antohisorin. I can help get this merged and release a new version. Dropped a comment

Updated. Sorry I don't know why I had duplicate values there.

izaaz commented:
Thanks @antohisorin . I'll get this merged and deployed once the tests pass.
